util/src/main/java/io/camunda/zeebe/util/logging/ThrottledLogger.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.util.logging;
+
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+
+/**
+ * Logger that logs only once every configured interval.
+ *
+ * <p>Example:
+ *
+ * <pre>
+ * while(condition) {
+ *     log.info("repeated log");
+ * }
+ * </pre>
+ *
+ * When ThrottledLogger is used, this is not logged in every loop. Instead, it will be logged once
+ * in every configured interval.
+ */
+public class ThrottledLogger implements Logger {
```

Comment:
:+1: I like the idea, it's something we could definitely use more often.

:wrench: I would allow passing a clock interface (whether the `Clock` standard class, or to be compatible with our actor framework, any long supplier, or even `InstantSource`), such that we can use it with our actor framework as expected...but it's also something we can do later, so take it or leave it

Reply:
That would be useful. I will defer it to later until we find use for this logger in other places.